### DISCUSSION TITLE
Github Graph

### DISCUSSION THREAD

**Discussion by Nurtimax** [Nurtimax/bilimgual#32] [Q&A] [answered]
how to create github graph

**Comment by Matraim** [comment, marked as answer]
1. yarn install react-activity-calendar 

Import Calendar from 'react-activity-calendar'

<Calendar 
data={arrayCalendarGrid}
theme={DEFAULT_THEME}
colorScheme="light"
blockSizee={10}
blockMargin={1}
/>

**Comment by Neomax05** [comment]
import React, { useEffect } from 'react';
import { Card, CardContent } from '@mui/material';
import Calendar from 'react-activity-calendar';

import { useAppDispatch, useAppSelector } from '../../../store/hooks';
import { DEFAULT_THEME, fillMissingDates } from '../../../utils/helpers/graph';
import { getDataThunk, selectorGithubGraph } from '../../../store/slices/github-graph';

const UserGraphChart = () => {
   const dispatch = useAppDispatch();
   const { data, year } = useAppSelector((state) => selectorGithubGraph(state));

   const arrayCalendarGrid = fillMissingDates(data, year);

   useEffect(() => {
      dispatch(getDataThunk());
   }, [dispatch]);

   return (
      <Card
         sx={{
            '& svg': {
               width: 'inherit',
               height: 'inherit'
            }
         }}
      >
         <CardContent>
            <Calendar
               data={arrayCalendarGrid}
               theme={DEFAULT_THEME}
               colorScheme="light"
               blockSize={10}
               blockMargin={1}
            />
         </CardContent>
      </Card>
   );
};

export default UserGraphChart;
